src/models/encryption-model.ts

```diff
@@ -0,0 +1,20 @@
+import { Model } from "sequelize";
+import { CryptorHttpClient, CryptorSecretKey  } from "../util/cryptor-http-client";
+import Logger from 'bunyan';
+
+export abstract class EncryptionModel<TSecretFields> extends Model{
+
+	abstract getEncryptContext(field: TSecretFields): Promise<Record<string, string|number>>;
+	abstract getCryptorKeyAlias(): CryptorSecretKey;
+
+	async decrypt(field: TSecretFields , logger: Logger) {
+		const ctx = await this.getEncryptContext(field);
+		return await CryptorHttpClient.decrypt(this.getDataValue[field as any], ctx, logger);
```

Comment:
Not from this PR, but in general: The `CryptorClient` should actually be renamed to `EncryptionClient` because the fact that it's using Cryptor under the hood is an implementation detail.